tests/lib/rules/no-array-constructor.js

```diff
@@ -47,76 +58,51 @@ ruleTester.run("no-array-constructor", rule, {
 	invalid: [
 		{
 			code: "new Array()",
+			output: "[]",
 			errors: [
 				{
 					messageId: "preferLiteral",
 					type: "NewExpression",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[]",
-						},
-					],
 				},
 			],
 		},
 		{
 			code: "new Array",
+			output: "[]",
 			errors: [
 				{
 					messageId: "preferLiteral",
 					type: "NewExpression",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[]",
-						},
-					],
 				},
 			],
 		},
 		{
 			code: "new Array(x, y)",
+			output: "[x, y]",
 			errors: [
 				{
 					messageId: "preferLiteral",
 					type: "NewExpression",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[x, y]",
-						},
-					],
 				},
 			],
 		},
 		{
 			code: "new Array(0, 1, 2)",
+			output: "[0, 1, 2]",
 			errors: [
 				{
 					messageId: "preferLiteral",
 					type: "NewExpression",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[0, 1, 2]",
-						},
-					],
 				},
 			],
 		},
 		{
 			code: "const array = Array?.();",
+			output: "const array = [];",
```

Comment:
I don't think we should autofix here. The original code, `array` could end up being a new instance of `Array` or `undefined` while the fixed code can only ever be a new instance of `Array`.

Reply:
This rule already checks if the Array refers to a global built-in. See https://github.com/eslint/eslint/pull/19621#discussion_r2051461544

Reply:
I agree with @nzakas. `Array?.()` checks whether `Array` exists and returns `undefined` if it doesn't, and although it looks unusual to check whether the global builtin `Array` exists in the runtime, it seems safer to provide a suggestion instead of autofix in this case.

Reply:
@snitin315 it's technically possible for someone to do `globalThis.Array = null`, so we need to be safe here.

Reply:
Ah! Makes sense. Let's not autofix in this case then.